### PR TITLE
Add an option to drag'n'drop selected text in ``TextEdit``

### PR DESCRIPTION
**Pull request by ConteZero**
Should solve https://github.com/godotengine/godot-proposals/issues/4629

### CODE REVIEW

**Comment by YuriSizov** [comment]
Please amend your commit message to follow the preferred format, "Verb + more descriptive statement about what changed". See the updated PR title for example.

File: editor/code_editor.cpp [hunk 1]
```diff
@@ -1024,6 +1024,7 @@ void CodeTextEditor::update_editor_settings() {
 	text_editor->set_scroll_past_end_of_file_enabled(EditorSettings::get_singleton()->get("text_editor/behavior/navigation/scroll_past_end_of_file"));
 	text_editor->set_smooth_scroll_enabled(EditorSettings::get_singleton()->get("text_editor/behavior/navigation/smooth_scrolling"));
 	text_editor->set_v_scroll_speed(EditorSettings::get_singleton()->get("text_editor/behavior/navigation/v_scroll_speed"));
+	text_editor->set_drag_and_drop_selection_enabled(EditorSettings::get_singleton()->get("text_editor/behavior/navigation/drag_and_drop_selection"));
```
Comment (Paulb23) [inline editor/code_editor.cpp:1027]: Need to define this setting in `editor_settings.cpp`

File: scene/gui/text_edit.cpp [hunk 2]
```diff
@@ -4458,6 +4458,14 @@ bool TextEdit::is_scroll_past_end_of_file_enabled() const {
...
 }
 
+void TextEdit::set_drag_and_drop_selection_enabled(const bool p_enabled) {
+	drag_and_drop_selection_enabled = p_enabled;
+}
+
+bool TextEdit::is_drag_and_drop_selection_enabled() const {
+	return drag_and_drop_selection_enabled;
+}
+
```
Comment (Paulb23) [inline scene/gui/text_edit.cpp:4468]: `.h` and `.c` content should follow the same order, so this should be under `delete_selection` below, same with the other changes. Including moving `drag_and_drop_selection_enabled` under selection.

`Nitpick:` Would move it under `is_deselect_on_focus_loss_enabled`

**Review by Paulb23 — CHANGES_REQUESTED** [review]

File: editor/code_editor.cpp (hunk 1 above)
Comment (ConteZero) [reply editor/code_editor.cpp:1027]: done

File: scene/gui/text_edit.cpp (hunk 2 above)
Comment (ConteZero) [reply scene/gui/text_edit.cpp]: done

**Review by Paulb23 — APPROVED** [review]

File: doc/classes/TextEdit.xml [hunk 3]
```diff
@@ -956,6 +956,9 @@
 		<member name="deselect_on_focus_loss_enabled" type="bool" setter="set_deselect_on_focus_loss_enabled" getter="is_deselect_on_focus_loss_enabled" default="true">
 			If [code]true[/code], the selected text will be deselected when focus is lost.
 		</member>
+		<member name="drag_and_drop_selection" type="bool" setter="set_drag_and_drop_selection_enabled" getter="is_drag_and_drop_selection_enabled" default="true">
+			If [code]true[/code] allow drag and drop of selected text.
```
Comment (akien-mga) [inline doc/classes/TextEdit.xml:960]:
```suggestion
			If [code]true[/code], allow drag and drop of selected text.
```

File: scene/gui/text_edit.cpp [hunk 4]
```diff
@@ -5405,6 +5416,7 @@ void TextEdit::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "shortcut_keys_enabled"), "set_shortcut_keys_enabled", "is_shortcut_keys_enabled");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "selecting_enabled"), "set_selecting_enabled", "is_selecting_enabled");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "deselect_on_focus_loss_enabled"), "set_deselect_on_focus_loss_enabled", "is_deselect_on_focus_loss_enabled");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "drag_and_drop_selection"), "set_drag_and_drop_selection_enabled", "is_drag_and_drop_selection_enabled");
```
Comment (akien-mga) [inline scene/gui/text_edit.cpp:5419]:
```suggestion
	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "drag_and_drop_selection_enabled"), "set_drag_and_drop_selection_enabled", "is_drag_and_drop_selection_enabled");
```
To be consistent with other boom properties in the same class.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Pull request merged**